Searches/BinarySearch.java

```diff
@@ -45,7 +45,7 @@
     if (right < left) return -1; // this means that the key not found
 
     // find median
-    int median = (left + right) >>> 1;
+    int median = left + ((right-left) >>> 1);
```

Comment:
Actually this change is not necessary. Above expression is already using a [logical shift](https://en.wikipedia.org/wiki/Logical_shift) (three `>>>`) which will preserve the sign bit if a potential overflow happens when summing (left + right). For languages like C++ and others that only support [arithmetic shift](https://en.wikipedia.org/wiki/Arithmetic_shift) (two `>>`) above workaround is necessary.

TL;DR original code works fine